src/devices/sound/mas3507d.cpp

```diff
@@ -78,6 +78,21 @@ void mas3507d_device::device_start()
 	save_item(NAME(i2c_io_val));
 	save_item(NAME(i2c_sdao_data));
 	save_item(NAME(playback_status));
+
+	// This should be removed in the future if/when native MP3 decoding is implemented in MAME
+	save_item(STRUCT_MEMBER(mp3_dec, mdct_overlap));
+	save_item(STRUCT_MEMBER(mp3_dec, qmf_state));
+	save_item(STRUCT_MEMBER(mp3_dec, reserv));
+	save_item(STRUCT_MEMBER(mp3_dec, free_format_bytes));
+	save_item(STRUCT_MEMBER(mp3_dec, header));
+	save_item(STRUCT_MEMBER(mp3_dec, reserv_buf));
+
+	save_item(STRUCT_MEMBER(mp3_info, frame_bytes));
+	save_item(STRUCT_MEMBER(mp3_info, frame_offset));
+	save_item(STRUCT_MEMBER(mp3_info, channels));
+	save_item(STRUCT_MEMBER(mp3_info, hz));
+	save_item(STRUCT_MEMBER(mp3_info, layer));
+	save_item(STRUCT_MEMBER(mp3_info, bitrate_kbps));
```

Comment:
You don’t actually need to use `STRUCT_MEMBER` for these, `NAME(mp3_dec.mdct_overlap)` etc. will work fine.  `STRUCT_MEMBER` provides glue for saving a member of a struct within an array as a single safe state item.  It's for stuff like this:
```
struct foo_t {
    u8 x, y;
};
foo_t foo[16];
...
save_item(STRUCT_MEMBER(foo, x));
save_item(STRUCT_MEMBER(foo, y));
```

It makes members of the structure in an array look like arrays to the save state system and the debugger.  Since `mp3_dec` and `mp3_info` aren’t inside arrays, you don’t need the extra magic.